src/experimental/tools/videoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,224 @@
+import dash from "../../../transports/dash";
+import getContentInfos from "./get_content_infos";
+import {
+  disposeSourceBuffer,
+  initSourceBuffer$,
+} from "./init_source_buffer";
+import removeBuffer$ from "./remove_buffer";
+import { IContentInfos } from "./types";
+import VideoThumbnailLoaderError from "./video_thumbnail_loader_error";
+
+const PPromise = typeof Promise === "function" ? Promise :
+                                                 pinkie;
+
+interface IJob { contentInfos: IContentInfos;
+                 segment: ISegment;
+                 stop: () => void;
+                 jobPromise: Promise<unknown>; }
+
+const segmentLoader = createSegmentLoader(
+  dash({ lowLatencyMode: false }).video.loader,
+  { maxRetry: 0,
+    maxRetryOffline: 0,
+    initialBackoffDelay: 0,
+    maximumBackoffDelay: 0, }
+);
+
+/**
+ * This tool, as a supplement to the RxPlayer, intent to help creating thumbnails
+ * from a video source.
+ *
+ * The tools will extract a "thumbnail track" either from a video track (whose light
+ * chunks are adapted from such use case) or direclty from the media content.
+ */
+export default class VideoThumbnailLoader {
+  private readonly _videoElement: HTMLVideoElement;
+
+  private _player: Player;
+  private _currentJob?: IJob;
+
+  constructor(videoElement: HTMLVideoElement,
+              player: Player) {
+    this._videoElement = videoElement;
+    this._player = player;
+  }
+
+  /**
+   * Set time of thumbnail video media element :
+   * - Remove buffer when too much buffered data
+   * - Search for thumbnail track element to display
+   * - Load data
+   * - Append data
+   * Resolves when time is set.
+   * @param {number} time
+   * @returns {Promise}
+   */
+  setTime(time: number): Promise<unknown> {
+    for (let i = 0; i < this._videoElement.buffered.length; i++) {
+      if (this._videoElement.buffered.start(i) <= time &&
+          this._videoElement.buffered.end(i) >= time) {
+        this._videoElement.currentTime = time;
+        log.debug("VTL: Thumbnail already loaded.");
+        return PPromise.resolve(time);
+      }
+    }
+
+    const manifest = this._player.getManifest();
+    if (manifest === null) {
+      return PPromise.reject(new VideoThumbnailLoaderError("NO_MANIFEST",
+                                                           "No manifest available."));
+    }
+    const contentInfos = getContentInfos(time, manifest);
+    if (contentInfos === null) {
+      return PPromise.reject(new VideoThumbnailLoaderError("NO_TRACK",
+                                                           "Couldn't find track for this time."));
+    }
+    const initURL = contentInfos.representation.index.getInitSegment()?.mediaURLs?.[0] ?? "";
+    if (initURL === "") {
+      return PPromise.reject(new VideoThumbnailLoaderError("NO_INIT_DATA", "No init data for track."));
+    }
+    const segment = contentInfos.representation.index.getSegments(time, time + 10)[0];
```

Comment:
:warning: the second argument of `getSegments` is the duration, not the end time.

I don't know what would happen when doing `getSegments(time, 0)` and its true that we might want to display something in a case of a discontinuity. To see which logic is the best